flash/core/data/data_module.py

```diff
@@ -1152,3 +1153,101 @@ def from_fiftyone(
+            val_transform: The dictionary of transforms to use during validation which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            test_transform: The dictionary of transforms to use during testing which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            predict_transform: The dictionary of transforms to use during predicting which maps
+                :class:`~flash.core.data.process.Preprocess` hook names to callable transforms.
+            data_fetcher: The :class:`~flash.core.data.callback.BaseDataFetcher` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`.
+            preprocess: The :class:`~flash.core.data.data.Preprocess` to pass to the
+                :class:`~flash.core.data.data_module.DataModule`. If ``None``, ``cls.preprocess_cls``
+                will be constructed and used.
+            val_split: The ``val_split`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            batch_size: The ``batch_size`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            num_workers: The ``num_workers`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            sampler: The ``sampler`` argument to pass to the :class:`~flash.core.data.data_module.DataModule`.
+            preprocess_kwargs: Additional keyword arguments to use when constructing the preprocess. Will only be used
+                if ``preprocess = None``.
+
+        Returns:
+            The constructed data module.
+
+        Examples::
+
+            data_module = DataModule.from_labelstudio(
+                export_json='project.json',
+                data_folder='label-studio/media/upload',
+                val_split=0.8,
+            )
+        """
+        # create preprocess
+        preprocess = preprocess or cls.preprocess_cls(
+            train_transform,
+            val_transform,
+            test_transform,
+            predict_transform,
+            **preprocess_kwargs,
+        )
+        # loading export data
+        data_source = LabelStudioDataSource(
+            data_folder=data_folder,
+            export_json=export_json,
+            data_type=data_type,
+            backbone=preprocess_kwargs.get("backbone"),
+            max_length=preprocess_kwargs.get("max_length", 128),
+            clip_sampler=preprocess_kwargs.get("clip_sampler"),
+            clip_duration=preprocess_kwargs.get("clip_duration", 1),
+            decode_audio=preprocess_kwargs.get("decode_audio", False),
+            decoder=preprocess_kwargs.get("decoder", "pyav"),
+            multi_label=preprocess_kwargs.get("multi_label", False),
+        )
+        data_source.load_data()
+        # loading data sets
+        train_dataset, val_dataset, test_dataset, predict_dataset = data_source.to_datasets()
+
+        data = cls(
```

Comment:
Ideally we would just call `cls.from_data_source(DefaultDataSources.LABEL_STUDIO)` or something here